src/activity-logger/background/log-page-visit.js

```diff
@@ -52,17 +56,31 @@ export async function logPageVisit({ tabId, url }) {
     // The time to put in documents.
     const timestamp = Date.now()
 
-    // First create an identifier for the page being visited.
-    const storePageResult = await storePage({ tabId, url })
+    const pageId = generatePageDocId({ url })
+
+    const lookup = initSingleLookup()
+    const dbResult = await lookup(pageId)
+    const latestTime = dbResult == null ? null : dbResult.latest
+
+    let storePageResult
+
+    if (latestTime == null || timestamp - latestTime < threshold) {
```

Comment:
This whole condition + `latestTime` declaration could be simplified to something like:
```js
    const existingPage = await initSingleLookup()(pageId)

    let storePageResult
    if (existingPage == null || timestamp - existingPage.latest < threshold) {
        // First create an identifier for the page being visited.
        storePageResult = await storePage({ tabId, url })
    }
```